cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -787,9 +837,27 @@ fn gen_store_base_offset_multi<M: ABIMachineSpec>(
     ret
 }
 
+fn legalize_signature(sig: &ir::Signature) -> ir::Signature {
+    let params_structret = sig
+        .params
+        .iter()
+        .find(|p| p.purpose == ArgumentPurpose::StructReturn);
+    let rets_have_structret =
+        sig.returns.len() > 0 && sig.returns[0].purpose == ArgumentPurpose::StructReturn;
```

Comment:
Is it wanted to a function returning multiple values, the second of which would be a struct return, would trigger the insertion below? (in other words: should this be "any return is the struct return")

Reply:
Fixed -- it should match any return value's purpose.